### PR TITLE
plugins: expand `PluginStatusV1` to include runtime status details

### PR DESCRIPTION
**Pull request by tigrato**
This PR extends Teleport PluginStatusV1 to include details about the status of the runtime status of a plugin. These details include:
- error message
- last sync time
- number of objects imported in gitlab

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6074,6 +6074,28 @@ message PluginOAuth2AuthorizationCodeCredentials {
 // PluginStatus is the user-facing status for the plugin instance.
 message PluginStatusV1 {
   PluginStatusCode code = 1;
+  // error_message is a human-readable error message that can be displayed to the user.
+  string error_message = 2;
+  // last_sync_time is the last time the plugin was run.
+  google.protobuf.Timestamp last_sync_time = 4 [
```
Comment (justinas) [inline api/proto/teleport/legacy/types/types.proto:6080]: is field number 3 missing intentionally?

**Review by justinas — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (justinas) [reply api/proto/teleport/legacy/types/types.proto:6080]: Do we want to have "last successful sync time" + "last attempted sync (possibly failed) time"?

**Comment by justinas** [comment]
Generally looks good.

What will the semantics of status updates be? E.g. if a code path emits the old barebones status (just the error code), will it override existing detailed integration-specific fields?

**Comment by tigrato** [comment]
> Generally looks good.
> 
> What will the semantics of status updates be? E.g. if a code path emits the old barebones status (just the error code), will it override existing detailed integration-specific fields?

We have no way of controlling this. Since this runs in auth, auth itself is responsible for updating the status field. If it's not aware of this change, it will replace whatever is in the db

**Review by jakule — APPROVED** [review]

**Review by nklaassen — APPROVED** [review]

**Pull request merged**